6573061 Xscreensaver displays "Xlib: extension "Xfree86-Misc" missing on display"

Fix created by Adam Jackson <ajax@nwnk.net>, 
posted to xorg@lists.freedesktop.org in
http://lists.freedesktop.org/archives/xorg/2008-July/037270.html

diff -up xscreensaver-5.05/driver/lock.c.jx xscreensaver-5.05/driver/lock.c
--- xscreensaver-5.05/driver/lock.c.jx	2008-07-16 11:45:25.000000000 -0400
+++ xscreensaver-5.05/driver/lock.c	2008-07-16 11:55:14.000000000 -0400
@@ -1426,6 +1426,10 @@ xfree_lock_grab_smasher (saver_info *si,
 {
   saver_preferences *p = &si->prefs;
   int status;
+  int event, error;
+
+  if (!XF86MiscQueryExtension(si->dpy, &event, &error))
+    return;
 
   XErrorHandler old_handler;
   XSync (si->dpy, False);
